sandbox/cgroup/cgroup_test.go

```diff
@@ -236,44 +238,145 @@ func (s *cgroupSuite) TestProgGroupBadSelector(c *C) {
+	err := os.MkdirAll(path, 0755)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(filepath.Join(path, "cgroup.procs"), buf.Bytes(), 0644)
 	c.Assert(err, IsNil)
-	c.Assert(pids, DeepEquals, []int{123, 234, 567, 999})
+}
 
-	pids, err = cgroup.PidsInGroup(s.rootDir, "group1/group2")
+func (s *cgroupSuite) TestPidsOfSnapEmpty(c *C) {
+	// Not having any cgroup directories is not an error.
+	pids, err := cgroup.PidsOfSnap("pkg")
 	c.Assert(err, IsNil)
-	c.Assert(pids, DeepEquals, []int{123, 234, 567})
+	c.Check(pids, HasLen, 0)
+}
+
+func (s *cgroupSuite) TestPidsOfSnapUnrelatedStuff(c *C) {
+	// Things that are not related to the snap are not being picked up.
+	s.writePids(c, "udisks2.service", []int{100})
+	s.writePids(c, "snap..service", []int{101})
+	s.writePids(c, "snap..scope", []int{102})
+	s.writePids(c, "snap.*.service", []int{103})
+	s.writePids(c, "snap.*.scope", []int{104})
 
-	pids, err = cgroup.PidsInGroup(s.rootDir, "group.does.not.exist")
+	pids, err := cgroup.PidsOfSnap("pkg")
 	c.Assert(err, IsNil)
-	c.Assert(pids, IsNil)
+	c.Check(pids, HasLen, 0)
 }
 
-func (s *cgroupSuite) TestPidsBadInput(c *C) {
-	err := os.MkdirAll(filepath.Join(s.rootDir, "group1"), 0755)
+func (s *cgroupSuite) TestPidsOfSnapSecurityTags(c *C) {
+	// Pids are collected and assigned to bins by security tag
+	s.writePids(c, "system.slice/snap.$RANDOM.pkg.hook.configure.scope", []int{1})
+	s.writePids(c, "system.slice/snap.pkg.daemon.service", []int{2})
+
+	pids, err := cgroup.PidsOfSnap("pkg")
 	c.Assert(err, IsNil)
-	gPids := []byte(`123
-zebra
-567
-`)
-	err = ioutil.WriteFile(filepath.Join(s.rootDir, "group1/cgroup.procs"), gPids, 0755)
+	c.Check(pids, DeepEquals, map[string][]int{
+		"snap.pkg.hook.configure": {1},
+		"snap.pkg.daemon":         {2},
+	})
+}
+
+func (s *cgroupSuite) TestPidsOfInstances(c *C) {
+	// Instances are not confused between themselves and between the non-instance version.
+	s.writePids(c, "system.slice/snap.pkg_prod.daemon.service", []int{1})
+	s.writePids(c, "system.slice/snap.pkg_devel.daemon.service", []int{2})
+	s.writePids(c, "system.slice/snap.pkg.daemon.service", []int{3})
+
+	// The main one
+	pids, err := cgroup.PidsOfSnap("pkg")
 	c.Assert(err, IsNil)
+	c.Check(pids, DeepEquals, map[string][]int{
+		"snap.pkg.daemon": {3},
+	})
 
-	pids, err := cgroup.PidsInGroup(s.rootDir, "group1")
-	c.Assert(err, ErrorMatches, `cannot parse pid "zebra"`)
-	c.Assert(pids, IsNil)
+	// The development one
+	pids, err = cgroup.PidsOfSnap("pkg_devel")
+	c.Assert(err, IsNil)
+	c.Check(pids, DeepEquals, map[string][]int{
+		"snap.pkg_devel.daemon": {2},
+	})
+
+	// The production one
+	pids, err = cgroup.PidsOfSnap("pkg_prod")
+	c.Assert(err, IsNil)
+	c.Check(pids, DeepEquals, map[string][]int{
+		"snap.pkg_prod.daemon": {1},
+	})
+}
+
+func (s *cgroupSuite) TestPidsOfAggregation(c *C) {
+	// A single snap may be invoked by multiple users in different sessions.
+	// All of their PIDs are collected though.
```

Comment:
This is perhaps obvious, but perhaps making a code comment in an appropriate place about how it is pids in aggregate that are looked at.

Reply:
I had a look at this code here and at the documentation of PidsOfSnap and I'm not sure what to change. The documentation says:

> The return value is a snapshot of the pid of a given san, grouped by
> security tag. ...

I'm happy to iterate but please suggest something concrete. Did I read you right that you'd like to change the documentation side, not the test side?